src/structure/dds_cache.rs

```diff
@@ -323,10 +325,11 @@ impl TopicCache {
     let split_key = match self.min_keep_samples {
       History::KeepAll => keys.nth(self.max_keep_samples as usize).copied(),
       History::KeepLast { depth } => {
+        let first_index = depth.min(self.max_keep_samples) as usize;
         let last_index = self.max_keep_samples.saturating_sub(depth) as usize;
```

Comment:
Can we rename these to `min_keep_count` & `max_keep_count` or something similar? I think `first/last_index` is not very clear without already knowing what they mean.

Reply:
I am just kind of against `max_keep_count`, since it's not actually how many we keep, but rather how many we keep on top of the `min_keep`.

How do you feel about `first_valid_sample` and `last_valid_sample`?

Reply:
I think `first/last_valid_sample` implies that they will be present, but that depends on the amount actually present in `self.changes`. Introducing `valid/invalid` might also lead to confusion what it actually means.

Since the indices don't start at the same point, naming them with the same scheme might also be confusing.

How about `min_keep_count` & `last_sample_index`? That is not as clear as I'd like since `last_sample_index` starts at `min_keep_count` but it's good enough for me.

Reply:
Sounds like a good middle ground :+1: